### ISSUE TITLE
False positives for `no-unused-vars` for enum variants

### ISSUE THREAD

**Issue by kevincox** [sveltejs/eslint-plugin-svelte3#202]
```svelte
<script lang="ts">
	const enum State {
		OPEN,
		CLOSED,
	}
	let state = State.CLOSED;
</script>

<button on:click={() => state = State.OPEN}>
	Show Menu
</button>

{#if state == State.OPEN}
	<div>Open!</div>
{/if}
```

```
Test.svelte
   2:13  error    'State' is defined but never used                                         no-unused-vars
   3:3   error    'OPEN' is defined but never used                                          no-unused-vars
   4:3   error    'CLOSED' is defined but never used                                        no-unused-vars
  13:6   warning  Unnecessary conditional, both sides of the expression are literal values  @typescript-eslint/no-unnecessary-condition
```

---

- @typescript-eslint/eslint-plugin: 5.55.0
- @typescript-eslint/parser: 5.55.0
- eslint: 8.36.0
- eslint-plugin-svelte3: 4.0.0
- svelte: 3.57.0
- typescript: 4.9.5

```javascript
module.exports = {
	extends: [
		"eslint:recommended",
		"plugin:@typescript-eslint/strict",
	],
	parser: "@typescript-eslint/parser",
	parserOptions: {
		project: "tsconfig.json",
		extraFileExtensions: [".svelte"],
	},
	overrides: [{
		files: ["*.svelte"],
		processor: "svelte3/svelte3",
	}],
	plugins: [
		"@typescript-eslint",
		"svelte3",
	],
	settings: {
		"svelte3/typescript": true,
	},
};
```

**Comment by kevincox** [comment]
User error.

When you are using typescript the base `no-unused-vars` is known-broken and should be disabled in favour of [@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars).
